Core/Main/itkElastixRegistrationMethod.hxx

```diff
@@ -868,20 +868,20 @@ ElastixRegistrationMethod<TFixedImage, TMovingImage>::GetCombinationTransform()
 
 template <typename TFixedImage, typename TMovingImage>
 auto
-ElastixRegistrationMethod<TFixedImage, TMovingImage>::ConvertToItkTransform(const TransformType & elxTransform)
+ElastixRegistrationMethod<TFixedImage, TMovingImage>::ConvertToItkTransform(const TransformType * elxTransform)
   -> SmartPointer<TransformType>
```

Comment:
Following from other changes in this PR, shouldn't this also be a raw pointer instead of an `itk::SmartPointer`?

My understanding is that `itk::Transform` objects are `itk::LightObjects` and have their own internal reference counts. As a result it seems possible that wrapping with a `SmartPointer` is introducing a redundant reference count, though possibly with different management rules.

In the SWIG C++ generated wrapping layer only the underlying transform object is explicitly incremented regardless of whether the return type is a raw pointer or `itk::SmartPointer`. For comparison with proposed return type change to `GetCombinationTransform`:

`itk::SmartPointer` return type:
```cpp
SWIGINTERN PyObject *_wrap_itkElastixRegistrationMethodIF2IF2_GetCombinationTransform(PyObject *SWIGUNUSEDPARM(self), PyObject *args) {
  itkTransformD22_Pointer result;
...
result = ((itkElastixRegistrationMethodIF2IF2 const *)arg1)->GetCombinationTransform();
...
{
    // get the raw pointer from the smart pointer
    itkTransformD22 * ptr = result;
    // always tell SWIG_NewPointerObj we're the owner
    resultobj = SWIG_NewPointerObj((void *) ptr, SWIGTYPE_p_itkTransformD22, 1);
    // register the object, it it exists
    if (ptr) {
      ptr->Register();
    }
  }
  return resultobj;
```

Raw pointer return type:
```cpp
SWIGINTERN PyObject *_wrap_itkElastixRegistrationMethodIF2IF2_GetCombinationTransform(PyObject *SWIGUNUSEDPARM(self), PyObject *args) {
itkTransformD22 *result = 0 ;
...
result = (itkTransformD22 *)((itkElastixRegistrationMethodIF2IF2 const *)arg1)->GetCombinationTransform();
...
{
    // always tell SWIG_NewPointerObj we're the owner
    resultobj = SWIG_NewPointerObj((void *) result, SWIGTYPE_p_itkTransformD22, 1);
										
    if (result) {
      result->Register();
    }
  }
  return resultobj;
```

Reply:
Yes, for Python wrapping, we could just return a raw pointer. And the wrapping object will `Register` and `UnRegister` as partially observed in the generated code.

In pure C++, returning a SmartPointer on an `itk::LightObject` that was created in the function will [Register](https://github.com/InsightSoftwareConsortium/ITK/blob/1551ff3f2306346dae7b385680277e75580acf09/Modules/Core/Common/include/itkSmartPointer.h#L205) and [UnRegister](https://github.com/InsightSoftwareConsortium/ITK/blob/1551ff3f2306346dae7b385680277e75580acf09/Modules/Core/Common/include/itkSmartPointer.h#L103).